Tests/subscriptions.Tests.ps1

```diff
@@ -29,4 +28,141 @@ Describe 'Geos' {
+                return Get-Content -Path "./Tests/Data/subscriptions.json"
+            } 
+
+            throw "Unexpected command: $Command"
+        }
+    }
+
+    It 'Can successfully get matching subscription details' {
+        $subscriptionsJson = Get-Content -Path "./Tests/Data/accounts.json"
+        $subscriptions = Convert-LinesToObject $subscriptionsJson
+        $result = Get-SubscriptionDetailsUsingAppName -AppServiceName "app-service-name" -subscriptions $subscriptions
+        $expected = $subscriptions[0]
+        $result | Should -Be $expected
+    }
+
+    
+    It 'Will throw an error if no matching subscription exists in accounts' {
+        $subscriptionsJson = Get-Content -Path "./Tests/Data/accounts-no-match.json"
+        $subscriptions = Convert-LinesToObject $subscriptionsJson
+        { Get-SubscriptionDetailsUsingAppName -AppServiceName "app-service-name" -subscriptions $subscriptions 2>$null } | Should -Throw
+    }
+
+    It 'Will throw an error if graph query has no output' {
+        Mock Invoke-Az {
+            param($Command)
+
+            if ($Command[0] -eq "graph" -and $Command[1] -eq "query")
+            {
+                return $null
+            } 
+
+            throw "Unexpected command: $Command"
+        }
+
+        $subscriptionsJson = Get-Content -Path "./Tests/Data/accounts.json"
+        $subscriptions = Convert-LinesToObject $subscriptionsJson
+        { GetSubscriptionDetailsUsingAppName -AppServiceName "app-service-name" -subscriptions $subscriptions } | Should -Throw
+
+    }
+
+
+}
+
+Describe 'GetSubscriptionDetails' {
+    BeforeAll {
+        Mock az {
+            param($azCommand)
+
+            if ($azCommand -eq "account")
+            {
+                return Get-Content -Path "./Tests/Data/accounts.json"
+            } 
+            elseif ($azCommand -eq "graph")
+            {
+                return Get-Content -Path "./Tests/Data/subscriptions.json"
+            }
+
+            throw "Unexpected command: $azCommand"
+        }
+
+        Mock Invoke-Az {
+            param($azCommand)
+
+            if ($azCommand[0] -eq "graph" -and $azCommand[1] -eq "query")
+            {
+                return Get-Content -Path "./Tests/Data/subscriptions.json"
+            }
+
+            throw "Unexpected command: $azCommand"
+        }
+    }
+
+    It 'Can successfully get matching subscription details' {
+        $result = GetSubscriptionDetails -SubscriptionId "subscription-id"
+        $subscriptions = Convert-LinesToObject $(Get-Content -Path "./Tests/Data/accounts.json")
+        
+        $expected = $subscriptions[0]
+        # TODO: Work out why pester doesn't think these two objects are equal when they clearly are (without ["id"])
+        $result["id"] | Should -Be $expected["id"]
+    }
+
+    It 'Lets you select a subscription with the app service name flag' {
+        $result = GetSubscriptionDetails -SearchAllSubscriptions 1 -AppServiceName app-service-name

```

Comment:
You are passing "1" for -SearchAllSubscriptions here and on other places, while it is of type bool. 1 converts to true, but I think it is a better style to pass $true. (and $false instead of 0, but I haven't seen such a test)

Reply:
I think I was going to write a test with SearchAllSubscriptions false, but I couldn't work out how to simulate user input in the test. Do you know how I could go about this?

Reply:
It is using Read-Host, so it will be difficult, but here is an approach: https://stackoverflow.com/a/65698966/4054714

But it looks rather complicated. Let's better write tests for something simpler before doing this.

If we wanted to start, we could modify the program to just use the only subscription if there is only one. Then we could test that.